pkg/apis/monitoring/v1alpha1/types.go

```diff
@@ -269,3 +278,25 @@ type ThanosQuerierSpec struct {
 // It should always be reconstructable from the state of the cluster and/or outside world.
 type ThanosQuerierStatus struct {
 }
+
+// SecretKeySelector selects a key of a secret
+type SecretKeySelector struct {
+	// The name of the secret in the object's namespace to select from.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+	// The key of the secret to select from.  Must be a valid secret key.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Required
+	Key string `json:"key"`
+}
+
+// WebTLSConfig contains configuration to enable TLS on web endpoints
+type WebTLSConfig struct {
+	// Secret containing the TLS key for the server
+	Key SecretKeySelector `json:"key"`
+	// Secret containing the TLS certificate for the server
+	Cert SecretKeySelector `json:"cert"`
+	// Secret containing the CA certificate to verify the server certificates
+	CA SecretKeySelector `json:"ca"`
```

Comment:
Do we need this for Alertmanager?

Reply:
This depends. Do we want to support the case, where the TLS certificate for prometheus and the certificate for alertmanager is signed by a different CA? I don't see a reason why somebody would do that, but theoretically there is nothing wrong on such a configuration. But in that case, if we want to have the self-scraping jobs being able to scrape prometheus and alertmanager, we would need to know the CA cert of both.

If we decide, that we require the user to use a single CA for the prometheus and alertmanager web endpoints, then we would need the CA field only once.

Reply:
Yes, makes sense. Sorry for the noise.